pippy/IR.py

```diff
@@ -1207,6 +1208,27 @@ def __str__(self):
     def __repr__(self):
         return self.split_gm.__repr__()
 
+    @dataclass
+    class PipeInfo:
+        graph: fx.Graph
+        num_stages: int
+        num_chunks: int
+        has_loss_and_backward: bool
+        args_chunk_spec: Optional[Tuple[Any, ...]] = None
+        kwargs_chunk_spec: Optional[Dict[str, Any]] = None
+        output_chunk_spec: Any = None  # TODO (kwen2501): type it better
+
+    def info(self) -> PipeInfo:
```

Comment:
why do you need this method, that seems to take an existing PipeInfo and make it into a new one?

Reply:
? 
`self` here is a `Pipe`, not `PipeInfo`

Reply:
oh! missed that